src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala

```diff
@@ -0,0 +1,70 @@
+package com.yotpo.metorikku.code.steps
+
+import com.yotpo.metorikku.exceptions.MetorikkuException
+import org.apache.log4j.{LogManager, Logger}
+import org.apache.spark.sql.{Column, DataFrame}
+import org.apache.spark.sql.functions._
+
+object SelectiveMerge {
+  val message = "You need to send 3 parameters with the names of the dataframes to merge and the key(s) to merge on" +
+    "(merged df1 into df2 favouring values from df2): df1, df2, Seq[String]"
+
+  private val log: Logger = LogManager.getLogger(this.getClass)
+
+  private val colRenamePrefix = "df2_"
+
+  def run(ss: org.apache.spark.sql.SparkSession, metricName: String, dataFrameName: String, params: Option[Map[String, String]]): Unit = {
+    params match {
```

Comment:
you can match case on all 3 params you're expecting, you might get some but not all

Reply:
and maybe use a single match case for this whole section, including the warnings/exceptions

Reply:
Done